mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp

```diff
@@ -101,6 +101,16 @@ convertAccessorImplDeviceType(sycl::AccessorImplDeviceType type,
                          type.getBody(), converter);
 }
 
+/// Converts SYCL accessor subscript type to LLVM type.
+static Optional<Type>
+convertAccessorSubscriptType(sycl::AccessorSubscriptType type,
+                             LLVMTypeConverter &converter) {
+  return convertBodyType(
+      "class.sycl::_V1::detail::accessor_common.AccessorSubscript." +
+          std::to_string(type.getCurrentDimension()),
+      type.getBody(), converter);
```

Comment:
If we want to mimic current clang output: They include the name of the type this type is a member of, but not the number of dimensions (as in your example).

Reply:
It does not matter for codegen (correctness) purposes. Printing the human readable form to match clang would only help to compare codegen between the 2 implementations. In this I think the codegen is clear enough.